cardinal/evm/server.go

```diff
@@ -285,11 +286,12 @@ func (s *msgServerImpl) getSignerComponentForAuthorizedAddr(
 	wCtx := ecs.NewReadOnlyWorldContext(s.world)
 	q, err := wCtx.NewSearch(ecs.Exact(ecs.SignerComponent{}))
 	if err != nil {
-		return nil, err
+		return nil, eris.Wrap(err, "error creating search")
 	}
-	err = q.Each(wCtx, func(id entity.ID) bool {
+	err = errors.Join(err, q.Each(wCtx, func(id entity.ID) bool {
 		var signerComp *ecs.SignerComponent
 		signerComp, err = component.GetComponent[ecs.SignerComponent](wCtx, id)
+		err = eris.Wrapf(err, "error getting signer component with id %d", id)
```

Comment:
we don't need to wrap the error here, we aren't using it anywhere

Reply:
You're right, the error doesn't need to be wrapped because it's already wrapped inside of `GetComponent`

However this error is used. It's subtle. err comes from the outside scope. It's set here and then joined with the top level error generated from q.Each